internal/cmd/keepassxctemplatefuncs.go

```diff
@@ -329,3 +354,82 @@ func (c *Config) keepassxcClose() error {
 	}
 	return nil
 }
+
+// keepassxcBuiltinExtractValues extract builtin values
+func (c *Config) keepassxcBuiltinExtractValues(
```

Comment:
This implementation reads all values from the keepass database to reduce access to the file itself.
If this might blow the memory I can change the implementation to only cache the requested values.

Reply:
We don't know how large any given keepass database will be, so the memory is potentially uncontrollable this way. More problematic, though, is that this decrypts the entire database and puts it in memory. chezmoi doesn't do anything specific to harden its memory accesses (I’m not sure what one could do with Go), so this represents a much larger attack surface on a running instance of chezmoi.

If it isn't too much additional work, I think that only entries requested should be cached because to do otherwise would change chezmoi's security profile.

Reply:
@halostatue @twpayne I've modified the code to only cache required entries